### ISSUE TITLE
Implement Parallel Processing for Performance Improvement

### ISSUE THREAD

**Issue by Shadows97** [Shadows97/PySubtitle#7]
The current processing of audio and video files in PySubtitle is sequential, which can be slow for large files. Implementing parallel processing can significantly improve performance.

**Tasks:**

- Investigate and implement parallel processing techniques for audio extraction and speech recognition.
- Ensure that the parallel processing implementation is thread-safe and efficient.
- Update the documentation to include information on performance improvements.

**Comment by hcm444** [comment]
concurrent.futures is probably an easy way

**Comment by Shadows97** [comment]
Yes but have already add it on your last pull req. It will be good if each new addition to the code is made on a branch associated with the issue concerned. This allows better visibility
